iroh-resolver/src/unixfs_builder.rs

```diff
@@ -671,14 +672,20 @@ mod tests {
 
     /// Read a stream of (cid, block) pairs into an in memory store and return the store and the root cid
     async fn stream_to_resolver(
-        stream: impl Stream<Item = Result<(Cid, Bytes)>>,
+        stream: impl Stream<Item = Result<Block>>,
     ) -> Result<(Cid, Resolver<Arc<fnv::FnvHashMap<Cid, Bytes>>>)> {
         tokio::pin!(stream);
         let items: Vec<_> = stream.try_collect().await?;
-        let (root, _) = items.last().context("no root")?.clone();
-        let store: fnv::FnvHashMap<Cid, Bytes> = items.into_iter().collect();
+        let root_block = items.last().context("no root")?.clone();
+        let store: fnv::FnvHashMap<Cid, Bytes> = items
+            .into_iter()
+            .map(|block| {
+                let (cid, bytes, _) = block.into_parts();
```

Comment:
Probably should have a test that checks that the links are actually stored in the store

Reply:
Will do. Just want to see how I like the Block struct. I think it is good...